akka-actor/src/main/scala/akka/actor/dungeon/DeathWatch.scala

```diff
@@ -81,15 +81,16 @@ private[akka] trait DeathWatch { this: ActorCell =>
           watching = removeFromMap(actor, watching)
         }
         if (!isTerminating) {
-          self.tell(optionalMessage.getOrElse(Terminated(actor)(existenceConfirmed, addressTerminated)), actor)
-          terminatedQueuedFor(actor)
+          self.tell(Terminated(actor)(existenceConfirmed, addressTerminated), actor)
```

Comment:
This looks more correct than before since it follows the same code path all the way as the `Terminated` message, e.g. there could have been an unwatch inbetween here and the actual delivery of the custom message.